packages/mongodb-dialects/java-driver/src/test/kotlin/com/mongodb/jbplugin/dialects/javadriver/glossary/JavaDriverDialectParserTest.kt

```diff
@@ -132,4 +161,343 @@ public final class Repository {
 
         assertEquals(HasCollectionReference.Unknown, unknownReference)
     }
+
+    @ParsingTest(
+        fileName = "Repository.java",
+        value = """
+import com.mongodb.client.MongoCollection;
+import com.mongodb.client.model.Filters;
+import org.bson.Document;
+import org.bson.types.ObjectId;
+import com.mongodb.client.FindIterable;
+
+public final class Repository {
+    private final MongoCollection<Document> collection;
+
+    public Repository(MongoCollection<Document> collection) {
+        this.collection = collection;
+    }
+
+    public FindIterable<Document> findBookById(ObjectId id) {
+        return this.collection.find(Filters.eq("_id", id));
+    }
```

Comment:
Ahh now I understand! I was wondering when would filter be a callable expression with key and value pairs. Totally forgot that this is quite common when writing Java.

Reply:
Yeah all key/value operators have the same syntax in the driver, that is really useful to just implement the pattern once without needing to check the specific operator.